src/libs/resizable_panel.js

```diff
@@ -0,0 +1,170 @@
+const sizes = {
+  0: window.innerHeight,
+  [window.innerHeight / 2]: window.innerHeight / 2,
+  [window.innerHeight]: 50,
+};
+
+const REDUCED = 'reduced';
+const FULL = 'full';
+const resizerClassName = 'deployable__list_resizer';
+export const resizableClassName = 'resizable_panel';
+
+function getClosest(arr, goal) {
+  return arr.reduce((prev, curr) =>
+    Math.abs(curr - goal) < Math.abs(prev - goal) ? curr : prev
+  );
+}
+
+export default class ResizablePanel {
+  constructor() {
+    this.timer = null;
+    this.panel = null;
+    this.holding = false;
+    this.isTransitioning = false;
+
+    this.checkPanel = this.checkPanel.bind(this);
+    this.holdResizer = this.holdResizer.bind(this);
+    this.move = this.move.bind(this);
+    this.stopResize = this.stopResize.bind(this);
+    this.playTransition = this.playTransition.bind(this);
+
+    window.deployedListHold = this.holdResizer;
+    window.deployedListStop = this.stopResize;
+  }
+
+  /**
+   * Looks for a resizable element
+   */
+  checkPanel() {
+    if (!this.panel) {
+      const element = document.querySelector(`.${resizableClassName}`);
```

Comment:
What happens if we have multiple resizable elements? Wouldn't it be better to set an id instead?

Reply:
It can't be. We either display the favorite panel or the category panel, never both. This feature was intended solely for the main panels and not for other elements, so i don't see the need to handle this use case.
I can see the benefit of using an id. I'll do the change 👍